### DISCUSSION TITLE
Running HAOS in a vm, with Asterisk Addon in a docker container = connection issues?

### DISCUSSION THREAD

**Discussion by zenlord** [TECH7Fox/asterisk-hass-addons#281] [Q&A] [answered]
Hi,
I've now been battling setting up this addon in my home, with two doorphones, one Android tablet and one Android phone, and I'm unsuccessful. I have set up a few Asterisk servers on bare metal before, but I'm afraid that I'm seeing connection issues, and maybe anyone here has any tips to improve the connection from the host network with Asterisk-in-a-container-in-a-virtualmachine?

The virtualmachine setup is a LXD vm with a managed bridge (lxdbr0 bridging my 192.168.2.14 address to 10.66.71.0/24)
My gateway/router has been configured to route 10.6.71.0/24 via 192.168.2.14

If I nmap the 5060 port on 10.66.71.35 from the host network, the port is listed as 'open'.

The symptoms I see is that when I dial from my doorphone, it rings 10-15 seconds, then connects to Asterisk, executes the dialplan and stops just before the Dial() or Queue() application:

```
    -- Removed contact 'sip:211@192.168.2.20:52466;transport=TCP;line=4212f7dc247ffed' from AOR '211' due to expiration
  == Contact 211/sip:211@192.168.2.20:52466;transport=TCP;line=4212f7dc247ffed has been deleted
  == Endpoint 211 is now Unreachable
```

-> The above 3 lines are triggered by hanging up from the doorpost (sip:211) GUI.

```
    -- Executing [500@doorpost:1] Answer("PJSIP/211-0000002b", "") in new stack
    -- Executing [500@doorpost:2] NoOp("PJSIP/211-0000002b", "## Call during office hours ##") in new stack
    -- Executing [500@doorpost:3] Answer("PJSIP/211-0000002b", "") in new stack
    **-- Auto fallthrough, channel 'PJSIP/211-0000002b' status is 'UNKNOWN'**
```

-> These 4 lines are triggered by pushing the button on my doorpost, configured to enter via context "doorpost" to extension "500". The dialplan has been reduced to the bare necessary and just contains another Dial() application, which is not executed. I have emphasized the last line.

```
  == Contact 212/sip:212@192.168.2.21:50639;transport=TCP;line=fd19834474fbb07 has been deleted
    -- Removed contact 'sip:212@192.168.2.21:50642;transport=TCP;line=659c798e2458483' from AOR '212' due to expiration
  == Contact 212/sip:212@192.168.2.21:50642;transport=TCP;line=659c798e2458483 has been deleted
  == Endpoint 212 is now Unreachable
    -- Executing [500@doorpost:1] Answer("PJSIP/212-0000002e", "") in new stack
    -- Executing [500@doorpost:2] NoOp("PJSIP/212-0000002e", "## Call during office hours ##") in new stack
    -- Executing [500@doorpost:3] Hangup("PJSIP/212-0000002e", "") in new stack
  **== Spawn extension (doorpost, 500, 3) exited non-zero on 'PJSIP/212-0000002e'**
    -- Added contact 'sip:212@192.168.2.21:50643;transport=TCP;line=659c798e2458483' to AOR '212' with expiration of 120 seconds
  == Endpoint 212 is now Reachable
```

-> This is another iteration of the same test, now from doorpost 212 (configured identically). The symptoms are the same: I push the button on the doorpost, hear it ring for 10-15 seconds, then the dialplan is executed but it fails.

More than happy to share configuration files, but my guess is that I'm overlooking something on the network level and I'm hoping someone here can just tell me "Hey, did you think of enabling XXX"? :).

**Comment by TECH7Fox** [comment, marked as answer]
Are the 10000-20000 ports open?

**Comment by zenlord** [comment]
No they were not - or at least, those ports were not open from the host network. The Asterisk container being inside a virtual machine that is managed with lxd kept giving me headaches, so I have installed a dedicated Asterisk LXD instance with networking through a routed nic, and now the headache is over.
Thank you for leading me to the solution (i.e. networking issues)!!
